src/lib/marketplace/MultisigCreateOrderStrategy.ts

```diff
@@ -0,0 +1,190 @@
+      subsetNonce: Number(orderDetails.subsetNonce),
+      orderNonce: orderDetails.orderNonce.toString(),
+      strategyId: Number(orderDetails.strategyId),
+      collectionType: orderDetails.collectionType,
+      collection: orderDetails.collection,
+      currency: orderDetails.currency,
+      signer: orderDetails.signer,
+      startTime: Number(orderDetails.startTime),
+      endTime: Number(orderDetails.endTime),
+      price: orderDetails.price.toString(),
+      itemIds: orderDetails.itemIds.map((id) => id.toString()),
+      amounts: orderDetails.amounts.map((amount) => Number(amount)),
+      additionalParameters: orderDetails.additionalParameters,
+      signature: "0x",
+      chainId: this.request.chainId,
+      id: "temporary",
+    };
+
+    const hec = new HypercertExchangeClient(
+      this.request.chainId,
+      // @ts-expect-error Typing issue with provider
+      EvmClientFactory.createEthersClient(this.request.chainId),
+    );
+
+    const [validationResult] = await hec.checkOrdersValidity([orderToValidate]);
+
+    if (!validationResult.valid) {
+      const errorCodes = validationResult.validatorCodes || [];
+
+      // Check if error codes follow the expected pattern. Everything needs to be 0 (valid),
+      // except for the signature validation error. This is because when this request is
+      // made, the message is missing one or more signatures.
+      // The signature will be validated in the command. It's only skipped for now.
+      // TODO: get the command name when ready
+      const isValidErrorPattern = errorCodes.every((code, index) => {
+        if (index === 3) {
+          return code === 421 || code === 0;
+        }
+        return code === 0;
+      });
+
+      // Only proceed if it's the expected signature validation error pattern
+      if (!isValidErrorPattern) {
+        throw new Errors.InvalidOrder(validationResult);
+      }
+    }
+    const tokenIds = orderDetails.itemIds.map(
+      (id) => `${this.request.chainId}-${orderDetails.collection}-${id}`,
+    );
+
+    const fractions = await Promise.all(
+      tokenIds.map((fractionId) => getFractionsById(fractionId)),
+    );
+
+    // Check if all fractions exist
+    if (fractions.some((fraction) => !fraction)) {
+      throw new Errors.MissingFractions();
+    }
+  }
+
+  private formatHypercertId(orderDetails: OrderDetails): string {
+    const tokenId = orderDetails.itemIds[0];
+    const hypercertTokenId = getHypercertTokenId(BigInt(tokenId));
+    const formattedHypercertId = `${this.request.chainId}-${orderDetails.collection}-${hypercertTokenId.toString()}`;
+    return formattedHypercertId;
+  }
+
+  private async createSignatureRequest(
+    messageHash: string,
+    safeAddress: string,
+    orderDetails: OrderDetails,
+  ): Promise<void> {
+    // Convert bigint values to strings for JSON serialization
+    const messageForStorage = {
+      ...orderDetails,
+      hypercert_id: this.formatHypercertId(orderDetails),
```

Comment:
@pheuberger  @Jipperism is this `hypercert_id` needed in the order/signature? The item Ids are what's being transfered no?